rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -188,50 +187,106 @@ static int create_enclave_range(int dev_fd, uint64_t mmap_size,
 	return 0;
 }
 
-static bool encl_create(int dev_fd, unsigned long bin_size,
-			struct sgx_secs *secs, struct enclave_info *encl_info,
-			struct metadata *meta_data,
-			struct sgx_sigstruct *sigstruct)
+/* Sanity check attributes to prevent possible tampering */
+static bool check_sgx_attributes(const uint64_t req_attrs,
+				 const uint64_t req_attrs_mask)
 {
-	struct sgx_enclave_create ioc;
-	uint64_t xfrm;
+	/* skeleton doesn't support 32-bit mode */
+	uint64_t enforced_pattern = SGX_ATTR_MODE64BIT;
 
-	memset(secs, 0, sizeof(*secs));
-	secs->attributes = SGX_ATTR_MODE64BIT;
-	if (enclave_debug)
-		secs->attributes |= SGX_ATTR_DEBUG;
-	/* Check attributes to prevent possible tampering of metadata area */
-	if ((meta_data->attributes & sigstruct->body.attributes) !=
-	    sigstruct->body.attributes) {
-		fprintf(stderr, "Invalid attributes value.\n");
+	if (req_attrs & ~SGX_ATTR_ALLOWED_MASK) {
+		fprintf(stderr,
+			"Invalid attributes value. The unsupported attributes %#lx are set.\n",
+			req_attrs & ~SGX_ATTR_ALLOWED_MASK);
+		return false;
+	}
+
+	if (req_attrs_mask & ~SGX_ATTR_ALLOWED_MASK) {
+		fprintf(stderr,
+			"Invalid attributes mask value. The unsupported attributes %#lx are set.\n",
+			req_attrs_mask & ~SGX_ATTR_ALLOWED_MASK);
 		return false;
 	}
-	secs->attributes = meta_data->attributes & secs->attributes;
-	/* Check the attributes in signature structure restrictions */
-	if ((sigstruct->body.attributes & sigstruct->body.attributes_mask) !=
-	    (secs->attributes & sigstruct->body.attributes_mask)) {
+
+	if ((req_attrs & enforced_pattern) != enforced_pattern) {
+		fprintf(stderr,
```

Comment:
May be `if (!(req_attrs & enforced_pattern)) ` is better because you just want to check SGX_ATTR_MODE64BIT bit?

Reply:
It is only correct when enforced_pattern == SGX_ATTR_MODE64BIT.

Using current logic can simplify the work once we need more bits set in enforced_pattern.

Reply:
Sound make sense.